File: AnagramFinder/WordReader.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;

namespace AnagramFinder
{
    public class WordReader {
        public string[] readWords(string path){
            var words = new List<string>();
            using (var sr = new StreamReader(path))
            {
                var line = sr.ReadLine();

                while (line != null)
                {
                    words.Add(removeSpaces(line));
                    line = sr.ReadLine();
                }
            }

               
            return words.ToArray();
        }

        private string removeSpaces(string word)
        {
            return word.Replace(" ", "");
        }
    }
}

File: src/WordReader.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class WordReader {
    public String[] readWords(String path){
        List<String> words = new ArrayList<>();
        BufferedReader br = null;
        try {
            br = new BufferedReader(new FileReader(path));
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        try {

            String line = br.readLine();

            while (line != null) {
                words.add(removeSpaces(line));
                line = br.readLine();
            }

            br.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return words.toArray(new String[words.size()]);
    }
    private String removeSpaces(String word) {
        return word.replace(" ", "");
    }

}
